WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -229,18 +229,24 @@ public TransactionFactory(Network network, KeyManager keyManager, ICoinsView coi
 			builder = builder.AddKeys(signingKeys.ToArray());
 			builder.SignPSBT(psbt);
 
+			var isPayjoin = false;
+
 			// Try to pay using payjoin
 			if (payjoinClient is not null)
 			{
 				psbt = TryNegotiatePayjoin(payjoinClient, builder, psbt, changeHdPubKey);
+				isPayjoin = true;
 				psbt.AddKeyPaths(KeyManager);
 				psbt.AddPrevTxs(TransactionStore);
 			}
 
 			psbt.Finalize();
 			tx = psbt.ExtractTransaction();
 
-			builder.CoinFinder = (outpoint) => psbt.Inputs.Select(x => x.GetCoin()).Single(x => x?.Outpoint == outpoint)!;
+			if (isPayjoin)
```

Comment:
I would probably go with:

```suggestion
			if (payjoinClient is not null)
```